compiler/front/cli_reporter.nim

```diff
@@ -490,6 +490,10 @@ proc reportBody*(conf: ConfigRef, r: SemReport): string =
 
       result.add("; routine: ", r.symstr)
 
+    of rsemUnavailableLocation:
+      result = ("accessed location '$1' doesn't exist in the current " &
+               "compile-time context") % [r.ast.render]
```

Comment:
```suggestion
      result = ("accessed non-compile-time location '$1' within a " &
               "compile-time context") % [r.ast.render]
```

I believe this makes it slightly more actionable, and reads a bit better from the perspective of "what's wrong with this line?".

Reply:
Yep, I think that's better, thank you.

Reply:
Hm, on further thought, the message would be incorrect in the following case:
```nim
static:
  var x = 0
  static:
    echo x
```
Here `x` is a compile-time location, but its definition happens at a "later" time (i.e., the inner `static` statement is evaluated first) than its usage, meaning that the usage of `x` is invalid/disallowed.

Reply:
Ah, multi-staged compilation for the win. 😆 

It's more like the location doesn't _yet_ exist in the current compile-time context. Maybe the word 'yet' is the hint for the end programmer to think about timing/stages?

Reply:
Do you mean the "cannot evaluate at compile-time" one or the "accessed location doesn't exist"?

Reply:
The "accessed location doesn't exist" one